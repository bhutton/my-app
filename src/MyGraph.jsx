import React, {useState} from 'react';
import axios from 'axios';
import {Flex} from '@chakra-ui/core';
import {AgGridColumn, AgGridReact} from 'ag-grid-react';
import {Link} from 'react-router-dom';

/* import "ag-grid-community/dist/styles/ag-grid.css";
 * import "ag-grid-community/dist/styles/ag-theme-alpine.css";
 *  */

/* import 'ag-grid-community/dist/styles/ag-theme-alpine.css'; */

export function LinkMethod() {
    return <Link to="/link">fred@bloggs.com</Link>;
}

export function MyGraph() {
    const response = [
        {
            name: 'Fred',
            value: '1234'.replace(/\B(?=(\d{3})+(?!\d))/g, ','),
            address: 'My Address',
            email: 'fred@bloggs.com',
        },
    ];
    const [rowData, setRowData] = useState();
    const data = getData(setRowData);
    function onGridReady(params) {
        setGridApi(params.api);
        setGridColumnApi(params.columnApi);
    }
    console.log('data = ', rowData);
    console.log('data2 = ', JSON.stringify(data.da));
    return (
        <Flex role="navigation" flex="0 0 auto" height="100%" position="sticky">
            <AgGridReact
                onGridReady={onGridReady}
                rowData={response}
                suppressColumnVirtualisation={process.env.NODE_ENV === 'test'}
            >
                <AgGridColumn field="name" />
                <AgGridColumn field="value" />
                <AgGridColumn field="address" />
                <AgGridColumn setRowData />
            </AgGridReact>
            <Link to="/">test</Link>
        </Flex>
    );
}

async function getData(setRowData) {
    console.log('get data');
    return await axios.get('/greeting').then((res) => {
        console.log('res = ', res);
        console.log('res = ', res.data);
        setRowData(res.data);
    });
}
