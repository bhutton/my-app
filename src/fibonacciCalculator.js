export function fibonacciCalculator(num) {
    if (num === 1) return [0];

    const response = [0, 1];

    for (let i = 2; i < num; i++) {
        response.push(response[i - 1] + response[i - 2]);
    }

    return response;
}

export function fibonacciToString(num) {
    const fibonacciSequence = `[${fibonacciCalculator(num)}]`;
    return fibonacciSequence;
}
