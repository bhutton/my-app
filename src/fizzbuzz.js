export function calculateFizzBuzz(num) {
    let response = '';
    if (num % 3 === 0) response += 'fizz';
    if (num % 5 === 0) response += 'buzz';

    if (response === '') return num;
    return response;
}
