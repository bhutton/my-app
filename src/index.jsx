import React from 'react';
import ReactDOM from 'react-dom';
import { MyGraph } from './MyGraph.jsx';

ReactDOM.render(
    <MyGraph />,
    document.getElementById('root')
);
