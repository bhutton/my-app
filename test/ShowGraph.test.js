import React from 'react';
import {render, screen} from '@testing-library/react';
import {MyGraph} from '../src/MyGraph';
import '@testing-library/jest-dom';
import {rest} from 'msw';
import {setupServer} from 'msw/node';
import {MemoryRouter} from 'react-router-dom';
import {createMemoryHistory} from 'history';

const response = [
    {
        name: 'Fred',
        value: '1234'.replace(/\B(?=(\d{3})+(?!\d))/g, ','),
        address: 'My Address',
        email: 'fred@bloggs.com',
    },
];
const server = setupServer(
    rest.get('/greeting', (req, res, ctx) => {
        return req(ctx.json({token: 'mocked_user_token'}));
    }),
);

describe('When loading graph', () => {
    beforeAll(() => server.listen());
    afterEach(() => server.resetHandlers());
    afterAll(() => server.close());

    it('Displays something', () => {
        const history = createMemoryHistory();

        // mock push function
        history.push = jest.fn();

        const {container} = render(
            <MemoryRouter history={history}>
                <MyGraph />
            </MemoryRouter>,
        );

        expect(screen.getByText('Name')).toBeInTheDocument();
        expect(screen.getByText('Fred')).toBeInTheDocument();
        expect(screen.getByText('Value')).toBeInTheDocument();
        expect(screen.getByText('1,234')).toBeInTheDocument();
        expect(screen.getByText('Address')).toBeInTheDocument();
        expect(screen.getByText('My Address')).toBeInTheDocument();
        /* expect(screen.getByText('Email')).toBeInTheDocument(); */
        expect(screen.getByText('test').href).toBe('http://localhost/');
    });
});
