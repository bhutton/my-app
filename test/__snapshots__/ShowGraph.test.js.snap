// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When loading graph Displays something 1`] = `null`;
